### PR TITLE
(feat) Integrated addEvent form and hooked up flux data flow

### PR DESCRIPTION
**Pull request by rjatkinson2**

### CODE REVIEW

File: client/js/components/week/week.js [hunk 1]
```diff
@@ -9,9 +11,16 @@ var Week = React.createClass({
   getInitialState: function () {
     return weekData;
```
Comment (joshuaNewman10) [inline client/js/components/week/week.js:13]: I think we should start commenting that the weekData, monthData, dayData are temporary data stubs until we get our server up and running. I will do so in my next PR, would be great if all followed suit!

File: client/js/constants/constants.js [hunk 2]
```diff
@@ -1,10 +1,12 @@
-var AppConstants = {
-  ViewActionTypes: {
-
-  },
-  ServerActionTypes: {
+var keyMirror = require('keymirror');
```
Comment (joshuaNewman10) [inline client/js/constants/constants.js:1]: Does KeyMirror literally just mirror a key to the value?

**Comment by joshuaNewman10** [comment]
This generally looks really good. Will wait till kyle takes a look at it so that we can make sure we are all on the same page before pulling it in

File: client/js/constants/constants.js (hunk 2 above)
Comment (rjatkinson2) [inline client/js/constants/constants.js:1]: Yup exactly - swaps out null with the key!

File: client/js/components/week/week.js (hunk 1 above)
Comment (rjatkinson2) [inline client/js/components/week/week.js:13]: Good call I'll update to make it clear.

File: client/js/actions/view-action-creator.js [hunk 3]
```diff
@@ -1,9 +1,15 @@
...
+var ApiUtils = require('../utils/api-utils');
+var AppConstants = require('../constants/constants');
 
 var ViewActionCreator = {
-
+  addEvent: function (event) {
+    AppDispatcher.dispatch({
+      actionType: AppConstants.ViewActionTypes.ADD_EVENT
+    });
+    ApiUtils.addEvent(event);
```
Comment (kyleaowen) [inline client/js/actions/view-action-creator.js]: i think we sould pass ServerActionCreator.eventAdded as the callback to APIutils.addEvent. thoughts?

File: client/js/utils/api-utils.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
+var ServerActionCreator = require('../actions/server-action-creator');
+var weekData = require('../weekData');
+var _ = require('underscore');
+
+var ApiUtils = {
+  addEvent: function (event) {
```
Comment (kyleaowen) [inline client/js/utils/api-utils.js]: event though it's not the final implementation - if you're making other changes, it'd be cool if you added a callback parameter which would take the ServerActionCreate.eventAdded as the argument. I think it'll set up a better convention for our api-utils. thoughts?

**Comment by kyleaowen** [comment]
Need to change all of the instances of 'event' to 'gig'!

**Comment by rjatkinson2** [comment]
We should discuss an agreed upon name for events that aligns with the application's name, eventfull. The changes should be made application wide in a separate commit to make sure we catch everything.

**Comment by joshuaNewman10** [comment]
Awesome thanks for making these changes Ryan! All looks good. And I agree that tonight we should do one commit that just changes every single event to gig. It will be messy if we do it seperately at different points in time

**Comment by kyleaowen** [comment]
this looks good - we should talk about the flux way to initialize the data stores, we can make some changes later on. Also, the 'event' to 'gig' thing needs to happen asap

**Pull request merged**